Framework/PythonInterface/plugins/algorithms/SaveYDA.py

```diff
@@ -0,0 +1,316 @@
+from mantid.api import PythonAlgorithm, AlgorithmFactory, MatrixWorkspaceProperty, WorkspaceUnitValidator, \
+                       InstrumentValidator, FileProperty, FileAction, mtd
+from mantid.kernel import Direction, CompositeValidator
+from mantid.dataobjects import Workspace2D
+
+import yaml
+from yaml import Dumper
+
+from collections import OrderedDict
+
+import math
+
+
+class SaveYDA(PythonAlgorithm):
+    """ Save data in yaml/frida 2.0 format from a Workspace2D.
+    """
+
+    def category(self):
+        """Return category
+        """
+        return "DataHandling\\Text"
+
+    def name(self):
+        """Return name
+        """
+        return "SaveYDA"
+
+    def summary(self):
+        """Return summary
+        """
+        return "Save Workspace to a Frida 2.0 yaml format"
+
+    def PyInit(self):
+        """Declare properties
+        """
+        wsValidators = CompositeValidator()
+        # X axis must be a NumericAxis in energy transfer units.
+        wsValidators.add(WorkspaceUnitValidator("DeltaE"))
+        # Workspace must have an Instrument
+        wsValidators.add(InstrumentValidator())
+
+        self.declareProperty(MatrixWorkspaceProperty(name="InputWorkspace", defaultValue="", direction=Direction.Input,
+                             validator=wsValidators), doc="Workspace name for input")
+        self.declareProperty(FileProperty(name="Filename", defaultValue="", action=FileAction.Save, extensions=""),
+                             doc="The name to use when writing the file")
+
+    def validateInputs(self):
+        """Basic validation for inputs.
+        :return: issues with not valid Inputs in dictionary
+        """
+        issues = dict()
+        # Only MomentumTransfer is allowed
+        allowUn = "MomentumTransfer"
+        ws = self.getProperty("InputWorkspace").value
+        # Y axis must be either a SpectrumAxis or a NumericAxis in q units.
+        # workspace must be a Workspace2D
+        if ws:
+            ax = ws.getAxis(1)
+
+            if not ax.isSpectra() and ax.getUnit().unitID() != allowUn:
+                issues["InputWorkspace"] = "Y axis is not 'Spectrum Axis' or 'Momentum Transfer'"
+
+            if not isinstance(ws, Workspace2D):
+                issues["InputWorkspace"] = "Input Workspace is not a Workspace2D"
+
+        return issues
+
+    def PyExec(self):
+        """ Main execution body
+        """
+        # Properties
+        ws = mtd[self.getPropertyValue("InputWorkspace")]
```

Comment:
The same  `.value` method can be used for the workspaces to avoid having to access it via a string:

```
ws = self.getProperty("InputWorkspace").value
```